stan/math/prim/prob/exp_mod_normal_cdf.hpp

```diff
@@ -48,59 +48,55 @@ return_type_t<T_y, T_loc, T_scale, T_inv_scale> exp_mod_normal_cdf(
   scalar_seq_view<T_inv_scale> lambda_vec(lambda);
   size_t N = max_size(y, mu, sigma, lambda);
 
-  for (size_t n = 0; n < N; n++) {
-    if (is_inf(y_vec[n])) {
-      if (y_vec[n] < 0.0) {
-        return ops_partials.build(0.0);
-      }
+  for (size_t n = 0, size_y = size(y); n < size_y; n++) {
```

Comment:
[question] any reason you prefer the style of defining a local `size_y` variable and using that as opposed to just having the comparison be `n < size(y)` without having the temporary variable? (It's not a style that I'm used to.)

Reply:
(btw, that is really a question... I haven't seen that style a lot and it's just a style question.)

Thanks for tightening up the bounds here.

Reply:
That was to ensure that `size()` is evaluated only once (thinking about it, this is probably an R bias!).  If that's not an issue in C++ I can change it not to use a local.

Reply:
The concise way to do this in C++11 is with algorithms and iterators:

```
if (std::any_of(y_vec.begin(), y_vec.end(), [](const auto& x) { return is_inf(x) && x < 0; }))
  return ops_partials.build(0);
```

I don't know if our `scalar_seq_view` types have the appropriate iterators, though.